source/dofs/dof_tools.cc

```diff
@@ -1149,7 +1149,8 @@ namespace DoFTools
       = (dynamic_cast<const parallel::Triangulation<DoFHandlerType::dimension,DoFHandlerType::space_dimension> *>
          (&dof_handler.get_triangulation()) == 0
          ?
-         1
+           (1+*std::max_element (subdomain_association.begin (),
+                                          subdomain_association.end ()))
```

Comment:
Just a minor remark regarding safety: Are we sure that `subdomain_association` is not empty at this place (you are de-referencing a pointer here)? I guess yes, because we cannot have a `dof_handler` with zero degrees of freedom...